test/integration/test_mail.py

```diff
@@ -0,0 +1,166 @@
+import pytest
+import requests
+from typing import List
+from testcontainers.core.container import DockerContainer
+from autosubmit.notifications.mail_notifier import MailNotifier
+from autosubmit.job.job_common import Status
+from typing import cast
+
+SMTP_PORT = 1025
+API_PORT = 8025
```

Comment:
I think we should make these random. Wouldn't it make the test fail if it's executed in parallel in the same host, trying to use the same ports? If so, you can use a similar approach to `test_paramiko_platform.py`, using a different range to get some ports for your tests.